utils/redirectedPathLocale.ts

```diff
@@ -0,0 +1,8 @@
+import { type Locale } from '@/configs/i18n.config';
+
+export const redirectedPathName = (pathName: string, locale: Locale) => {
+  if (!pathName) return '/';
+  const segments = pathName.split('/');
+  segments[1] = locale;
```

Comment:
tại sao lại re-assign như vầy?

lỡ `pathname` là `/dashboard/posts/1234`, sau khi chạy qua cái này sẽ bị biến thành `/dashboard/en/1234` sao?

Reply:
Mặc định path của mình lúc nào cũng [locale]/dashboard/posts/1234
nên khi vào hàm redirectPathName thì : 
```
 pathName = /en/dashboard/posts/1234
```
sau khi chạy qua hàm này sẽ có dạng [new locale]/dashboard/posts/1234

Reply:
nhưng nó re-assign vô cái `[1]` mà? cái `pathname` có shape như thế nào?

```
> '/dashboard/posts/1234'.split('/')
[ '', 'dashboard', 'posts', '1234' ]
```
assign vô cái vị trí `1` là cái `dashboard` bị mất rồi? hay anh hiểu sai chỗ nào?

Reply:
Không phải là đưa path sai, mà là lúc em viết link tới chỗ nào đó, em không thể ép người ta thêm `locale` vào được, ví dụ:

```
<NextLink href="/dashboard/posts">Go to Post</NextLink>
```

Ở đây người ta chỉ muốn đi tới Posts trong dashboard, mình không thể ép sử dụng `locale` từ slug hoài được. Tại vì khi người ta chọn option đổi locale bằng domain, mình phải đi sửa hết toàn bộ url trong source code.

Trong tương lai, sẽ có khách chọn kiểu:

```
en: https://example.com/dashboard/posts
vi: https://example.vn/dashboard/posts
```

Cho nên kể cả hiện tại mình sử dụng dynamic slug `[locale]`, nhưng vẫn phải nghĩ tới trường hợp mình phải đổi thành `[domain]`, do đó các helpers khi làm việc phải làm sao cho việc thay đổi dễ hơn, tức là toàn bộ các `NextLink` hay `router.push()` không nên biết tới sự tồn tại của locale.